src/main/java/com/jfrog/ide/common/scan/ScanLogic.java

```diff
@@ -0,0 +1,33 @@
+package com.jfrog.ide.common.scan;
+
+import com.jfrog.ide.common.configuration.ServerConfig;
+import com.jfrog.ide.common.log.ProgressIndicator;
+import org.jfrog.build.extractor.scan.Artifact;
+import org.jfrog.build.extractor.scan.DependencyTree;
+
+import java.io.IOException;
+
+public interface ScanLogic {
+    /**
+     * Scan and cache components.
+     *
+     * @param indicator - Progress bar.
+     * @param quickScan - Quick or full scan.
+     * @return true if the scan completed successfully, false otherwise.
+     */
+    boolean scanAndCacheArtifacts(ServerConfig server, ProgressIndicator indicator, boolean quickScan, ComponentPrefix prefix, Runnable checkCanceled) throws IOException ;
```

Comment:
The first argument (```server```) is missing from the javadoc description.
The same goes for the implementations fo this method in sub-classes.